File: app/backend/src/schemas/firefighter_dashboard.py
from enums.report_status import ReportStatus
from enums.fire_danger import FireDanger
from typing import List
from datetime import datetime
from pydantic import BaseModel, ConfigDict

class NearbyFire(BaseModel):
    location_text: str
    distance: float
    time_ago: str
    status: ReportStatus

    model_config = ConfigDict(from_attributes=True)

class EnvironmentVariables(BaseModel):
    wind: float
    wind_dir: int # wind angle in degrees 
    temperature: float
    fire_danger: FireDanger
    humidity: float

class NearbyFiresList(BaseModel):
    data: List[NearbyFire]
    total: int

File: app/frontend/src/types/FirefighterDashboard.ts
import type { ReportStatus } from './Report';

export type FireDanger = 'low' | 'medium' | 'high' | 'very high';

export interface NearbyFire {
  id: string;
  reference_number: string;
  location_text: string;
  distance: number;
  time_ago: string;
  status: ReportStatus;
  lat: number;
  lng: number;
  boundary_radius: number;
}

export interface EnvironmentVariables {
  wind: number;
  wind_dir: number;
  temperature: number;
  fire_danger: FireDanger;
  humidity: number;
}

export interface NearbyFiresList {
  data: NearbyFire[];
  total: number;
}

export interface DashboardData {
  nearby_fires: NearbyFiresList;
  environment_variables: EnvironmentVariables;
}
